### DISCUSSION TITLE
PBF File not being built

### DISCUSSION THREAD

**Discussion by margonarimartin** [gis-ops/docker-valhalla#130] [Q&A] [answered]
I'm having trouble trying to run the service after adding a new PBF file to the `custom_files/` folder.

So from what I understood from the documentation: if I add a new PBF file to the folder, and after running the valhalla service it should build the tiles for that file, but it's not happening.

This is the output from what I'm running:

```
docker run -t -v $PWD/custom_files:/custom_files -p 8002:8002 --name valhalla_gis-ops_us ghcr.io/gis-ops/docker-valhalla/valhalla:latest

INFO: Running container with user valhalla UID 59999 and GID 59999.

find: '/custom_files/transit_tiles': No such file or directory
WARNING: Hash not found for: /custom_files/arkansas-latest.osm.pbf!
INFO: Found routing tiles with 'use_tiles_ignore_pbf' True.
INFO: Jumping directly to the tile loading!
WARNING: User valhalla is running with sudo privileges. Try building the image with a host user's UID & GID.
INFO: Found config file. Starting valhalla service!
```

And later some messages regarding Traffic tiles, but nothing regarding building the tiles for the new file.

Maybe I misunderstood something or I'm doing something wrong. I'd appreciate some help

**Comment by nilsnolde** [comment, marked as answer]
This is the hint:

> INFO: Found routing tiles with 'use_tiles_ignore_pbf' True.

You'll need to change that to `False` so it ignores the already built tiles. Also bear in mind: https://github.com/valhalla/valhalla/issues/3925

**Reply by margonarimartin** [reply]
Two things in regards to this:

1. Isn't it supposed that the default for `use_tiles_ignore_pbf ` is `False` ?
2. I've started running it again with `use_tiles_ignore_pbf=False` and it seems to be building. But in this case will it also rebuild new-york-latest.osm.pbf which was the one that was previously built?

I'm just trying to understand if there is a possibility to build tiles in some kind of _incremental way_, so let's say for the US I run valhalla with a specific state (like New York), and then I add a new PBF file to custom_files folder, and when restarting it will build the new file.

I'm basing my comments in this statement from the README:

> If you change the PBF files by either adding new ones or deleting any, Valhalla will build new tiles on the next restart unless told not to (e.g. setting use_tiles_ignore_pbf=True).

**Reply by nilsnolde** [reply]
> Isn't it supposed that the default for use_tiles_ignore_pbf  is False ?

Seems the docs were wrong, it's corrected.

> if there is a possibility to build tiles in some kind of incremental way

That's mostly not possible because of how graphs work. They need to be connected. That is if you have bordering datasets. If you don't, say you put NYC first & LA later, then no problem. But this image doesn't support this case out-of-the-box. You could still build LA into a different folder and merge the `valhalla_tiles`, that's fine too.
